modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,59 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
+				var image         = media[ i ],
+					media_details = image.media_details,
+					media_sources = media_details.sources,
+					sizes         = media_details.sizes,
+					sizes_keys    = Object.keys( sizes );
 
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
+				// If the full image has no JPEG version available, no sub-size will have JPEG available either.
+				if ( sizes.full && ! sizes.full.sources['image/jpeg'] ) {
 					continue;
 				}
 
-				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
+				var images = document.querySelectorAll( 'img.wp-image-' + image.id );
+
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
-					var srcset = images[j].getAttribute( 'srcset' );
+
+					var src = images[ j ].src;
+
+					// If there are no sizes, then replace src through sources, there is nothing more to replace.
+					if ( media_sources && ! sizes.full ) {
+						src = src.replace( media_sources['image/webp'].file, media_sources['image/jpeg'].file );
+						images[ j ].setAttribute( 'src', src );
+						break;
+					}
```

Comment:
There are a few issues here still:
* The `! sizes.full` check here doesn't look right: We don't care here whether there is no `full` size, it's about whether there are no sizes at all.
* We need to check whether the sources actually have `image/webp` and `image/jpeg` in them.
* We must not `break` here as we still need to process other potential `images`. So it needs to be `continue`.
* We must check here whether `src` is actually present before trying to modify it. (Even when it is not present though for whatever reason, we can still bail though, because `srcset` then definitely won't be present either.)

In summary:
```suggestion
					// If there are sources but no sizes, then attempt to replace src through sources. In that case, there is nothing more to replace.
					if ( media_sources && ! sizes_keys.length ) {
						// Only modify src if available and the relevant sources are set.
						if ( src && media_sources['image/webp'] && media_sources['image/jpeg'] ) {
							src = src.replace( media_sources['image/webp'].file, media_sources['image/jpeg'].file );
							images[ j ].setAttribute( 'src', src );
						}
						continue;
					}
```

Reply:
Thanks, Felix for the detailed review. This feedback is addressed.